tests/org.eclipse.ui.tests/Eclipse UI Tests/org/eclipse/ui/tests/dialogs/UIEditWorkingSetWizardAuto.java

```diff
@@ -57,8 +61,32 @@ private IWorkingSetPage getDefaultEditPage() {
 		return registry.getDefaultWorkingSetPage();
 	}
 
+	/**
+	 * Blocks the calling thread until autobuild completes.
+	 */
+	public static void waitForBuild() {
+		((Workspace) getWorkspace()).getBuildManager().waitForAutoBuild();
+	}
+
+	/**
+	 * Enables or disables workspace autobuild. Waits for the build to be finished,
+	 * even if the autobuild value did not change and a previous build is still
+	 * running.
+	 */
+	public static void setAutoBuilding(boolean enabled) throws CoreException {
+		IWorkspace workspace = getWorkspace();
+		if (workspace.isAutoBuilding() != enabled) {
+			IWorkspaceDescription description = workspace.getDescription();
+			description.setAutoBuilding(enabled);
+			workspace.setDescription(description);
+		}
+		waitForBuild();
+	}
+
+
 	@Test
 	public void testEditPage() throws Throwable {
+		setAutoBuilding(false);
```

Comment:
This leaves the environment after the test in an unexpected state, as autobuilding is expected to be turned on. You can either
1. put the complete test in a try/finally block and restore the old autobuilding state in the finally block, or
2. restore the old autobuilding state in a test teardown method.
Usually the latter is the better choice as it avoid to wrap a whole test method in a try/finally block.

Reply:
https://github.com/eclipse-platform/eclipse.platform.ui/commit/8bb3de749ce6ddc05828bb747dc2c84d2a031889

I opted for the second solution, by Overriding doSetUp() and doTearDown();